nucypher/config/node.py

```diff
@@ -116,18 +137,36 @@ def payload(self):
                             )
         return base_payload
 
+    def _generate_runtime_filepaths(self, commit=True) -> dict:
+        """Dynamically generate paths based on configuration root directory"""
+        if self.temp and commit and self.config_root is constants.UNINITIALIZED_CONFIGURATION:
+            raise self.ConfigurationError("Cannot pre-generate filepaths for temporary node configurations.")
+        filepaths = dict(config_root=self.config_root,
+                         keyring_dir=os.path.join(self.config_root, 'keyring'),
+                         known_nodes_dir=os.path.join(self.config_root, 'known_nodes'),
+                         known_certificates_dir=os.path.join(self.config_root, 'certificates'),
+                         known_metadata_dir=os.path.join(self.config_root, 'metadata'))
+        if commit:
```

Comment:
I keep seeing this, not really apparent what `commit` is a boolean for. Just that it sets some attrs?

Reply:
Yeah - I think I'm going to walk back this Idea a bit, but its so that you can peek at config filepaths before setting (and writing) them.